File: Week-4/Gold/PGM-1339.py
answer = 0
cnt = dict()
n = int(input())
 
for i in range(n):
    a = input().strip();
    digit = 1
    for j in range(len(a)-1,-1,-1): # 각 string의 알파벳 별로 자릿수 더해주기
        if a[j] in cnt:
            cnt[a[j]]+= digit
        else:
            cnt[a[j]] = digit
        
        digit *= 10

result = sorted(cnt.values(),reverse=True) 

m = 9
for i in range(len(result)): # 각 알파벳 별로 큰 순서대로 큰 숫자 부여해서 더해주기
    answer += m*result[i]
    m-=1

print(answer)

File: Week-4/Gold/PGM-1339.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <map>
using namespace std;


int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    map<char,int> cnt; //알파벳 별 나오는 자릿수 저장
    int n;
    string a;
    
    int answer = 0;
    cin >> n;
    
    for(int i = 0 ; i < n; i++){
        cin >> a;
        int digit = 1;
        for(int j = a.size()-1 ; j > -1 ; j--){ //각 string의 1의 자리부터 알파벳 별로 digit 저장
            if(cnt[a[j]] > 0){
                cnt[a[j]]+= digit;
            }else{
                cnt[a[j]] = digit;
            }
            digit *= 10;
        }
    }
    
    
    vector<int> result;
    
    for(auto it = cnt.begin(); it != cnt.end(); it++)
        result.push_back(it->second);

    sort(result.begin(),result.end(),greater<int>()); // 큰 순서대로 정렬
    int m = 9;
    for(int i = 0; i < result.size(); i++){ // 큰 순서대로 큰 숫자 부여
        answer += m*result[i];
        m--;
    }
    
    cout << answer << endl;
    return 0;
}
